sdk/core/core-http/lib/atomXmlOperationSpec.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { HttpOperationResponse } from "./httpOperationResponse";
+
+/**
+ * @ignore
+ * Type representing the JSON representation of XML request data
+ */
+export interface XMLRequestInJSON {
+  [key: string]: {
+    $: { xmlns: string };
+    [key: string]: any;
+  };
+}
+
+/**
+ * @ignore
+ * Type representing the JSON representation of XML response data
+ */
+export interface XMLResponseInJSON {
+  [key: string]: any;
+}
+
+export interface AtomXmlSerializer {
+  serialize(requestBodyInJson: any): XMLRequestInJSON;
```

Comment:
The type `XMLRequestInJSON` has restrictions that it should have a particular structure, for example, `$: { xmlns: string };`. This restriction is useful in the `serializeToAtomXmlRequest()` method in Service Bus, but I dont believe is useful so much in `core-http`. Does the internals of the  `convertJsonToAtomXml` method expect the `$` or `$: { xmlns: string };`?

The `AtomSerializationPolicy` deals with calling the appropriate serializer, do the json->xml & xml->json conversions and should be agnostic to the structure of the json output of serlialize(). Otherwise, we are saying that every serializer implementation should serialize the input to exactly this format.

Instead of `XMLRequestInJSON`, I would suggest using `object` here and keep both `XMLRequestInJSON` and `XMLResponseinJSON` in the Service Bus library
`object` is better than `any` while allowing any json object to pass through.
@chradek thoughts?

```suggestion
  serialize(requestBodyInJson: object): object;
```

Reply:
If we are using object/any, do we still need the types even?

Reply:
Also, about the overall comment - looks like we'll still need to export the Atom XML helpers from core-http, as we likely will be introducing another serializer for AuthorizationRules, as it's a settable property and is specific to each operation

Reply:
> If we are using object/any, do we still need the types even?

The type `XMLResponseinJSON` is not very useful, we can inline `[key: string]: object`. 
`XMLRequestInJSON` has more restrictions in it than `XMLResponseinJSON`. It is useful if we have any code where we try to access the `content["$"]` where `content` is of type `XMLRequestInJSON`.

From where I can see, values of this type is returned by the serialize() method and then passed to `convertJsonToAtomXml()` which in turn is very agnostic to the contents of `content`.

So, I am not seeing the benefits of this type either.

This feedback is mainly coming from the fact that `convertJsonToAtomXml()` and `convertAtomXmlToJson()` do not care for the type of the content passed to them as long as the former gets a json object and the latter gets a string

Reply:
Agree, updated - one problem with using "object" as a type is that doing following gives error on the `result` "Element implicitly has an 'any' type because type '{}' has no index signature."

```
let result: object;
...
...
result[ServiceBusConstants.ATOM_METADATA_MARKER] = ...
```

So having to mostly fallback to result/any as was the case initially.

Reply:
Then, I would use `[key: string]: object`